src/main/java/com/alysaa/geyserupdater/velocity/util/GeyserVelocityDownloader.java

```diff
@@ -0,0 +1,72 @@
+package com.alysaa.geyserupdater.velocity.util;
+
+import com.alysaa.geyserupdater.common.util.FileUtils;
+import com.alysaa.geyserupdater.common.util.GeyserProperties;
+import com.alysaa.geyserupdater.velocity.VelocityUpdater;
+
+import com.velocitypowered.api.proxy.Player;
+import com.velocitypowered.api.proxy.ProxyServer;
+
+import net.kyori.adventure.text.Component;
+
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+public class GeyserVelocityDownloader {
+
+    private static boolean downloadSuccess;
+
+    /**
+     * Downloads the most recent Geyser build. If enabled in the config, the server will also attempt to restart.
+     *
+     * @return true if the download was successful
+     */
+    public static boolean updateGeyser() {
+        VelocityUpdater plugin = VelocityUpdater.getPlugin();
+        ProxyServer server = plugin.getProxyServer();
+        Logger logger = plugin.getLogger();
+
+        // Download the file
+        server.getScheduler().buildTask(plugin, () -> {
+            String fileUrl = null;
+            try {
+                fileUrl = "https://ci.opencollab.dev/job/GeyserMC/job/Geyser/job/" + GeyserProperties.getGeyserGitPropertiesValueForPropertyKey("git.branch") + "/lastSuccessfulBuild/artifact/bootstrap/velocity/target/Geyser-Velocity.jar";
+            } catch (IOException e) {
+                logger.error("Failed to get the current Geyser build's Git branch!");
+                e.printStackTrace();
```

Comment:
there should be a 
```
downloadSuccess = false;
return;
```
here so that we don't try downloading with a null fileUrl

Reply:
Addressed in commit `5b4e9fa7b4aad55af9514accec6a2146d15c96e0`.